mne/cov.py

```diff
@@ -265,13 +265,18 @@ def read_cov(fname, verbose=None):
 # Estimate from data
 
 @verbose
-def make_ad_hoc_cov(info, verbose=None):
+def make_ad_hoc_cov(info, std=None, verbose=None):
     """Create an ad hoc noise covariance.
 
     Parameters
     ----------
     info : instance of Info
         Measurement info.
+    std : dict of ndarray | None
+        Standard_deviations of the diagonal elements. If dict, keys should be
+        `grad` for gradiometers, `mag` for magnetometers and `eeg` for EEG
+        channels. The ndarrays must have the same number of elements as the
+        sensors denoted by their keys.
```

Comment:
But our defaults aren't (n_chan,) arrays, but simply floats. Is there a reason for not allowing float values here too?

Reply:
None, I just didn't think about it. It should be allowed for the case of all sensors having the same standard deviation, if all agree.

Reply:
For consistency with other code these should just be a single float per channel type. If you want something more complex than a single float per channel type, then `ad_hoc_cov` becomes a misnomer as you are talking about something more like a diagonal empirical noise cov.

Reply:
Ok yes it's trhe. But this functionality is also very interesting. It would be nice to be able to create a more complicated covariance matrix. Also the possibility of adding other types of noise is pretty interesting, at least for my research. Could we implement something like that?

Reply:
Yes, we could add a:
```
class CovarianceArray(object)
    def __init__(data, info):
```
convenience class (similar to RawArray, EpochsArray, etc.) that takes a `ndarray` `data` of shape `(n_ch,)` for diagonal or `(n_ch, n_ch)` for full covariance matrices, and `info` and created a `Covariance` object from it. It should only be a few lines of code to do it.

Let's do it separately, though, to give people time to comment on the API and necessity. In the meantime it won't hold up progress on this.

Reply:
BTW you might already find that constructing a `Covariance` object yourself directly isn't too difficult.

Reply:
Great!